### PR TITLE
Nishant/add firebase notif

### PR DESCRIPTION
**Pull request by nisrulz**
Added Firebase Notification

### CODE REVIEW

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        if (remoteMessage != null) {
+            Map<String, String> payloadMap = remoteMessage.getData();
+            String bodyStr = "NA";
+
+            // Check if message contains a data payload.
+            if (payloadMap.size() > 0) {
+                Log.d(TAG, "Payload: ");
+                for (String key : payloadMap.keySet()) {
+                    Log.d(TAG, "Key: " + key + ", Value: " + payloadMap.get(key));
```
Comment (kenyee) [inline Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java]: wrap Log.d in BuildConfig.DEBUG so it gets removed from release builds.
Or use Timber...it's a much better logging system.

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+            // Check if message contains a notification payload.
+            if (remoteMessage.getNotification() != null) {
+                bodyStr = remoteMessage.getNotification().getBody();
+                Log.d(TAG, "Body: " + bodyStr);
+            }
+
+            // Show the notification here
+            NotificationUtils notificationUtils = new NotificationUtils(this);
+            notificationUtils.sendAndroidChannelNotification("Droidcon Boston", bodyStr, 101);
```
Comment (kenyee) [inline Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java]: should "Droidcon Boston" be in strings.xml?

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/NotificationUtils.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            channelId) {
+        Intent resultIntent = new Intent(this, MainActivity.class);
+        PendingIntent pi = PendingIntent.getActivity(this, 0, resultIntent, PendingIntent
+                .FLAG_UPDATE_CURRENT);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext(),
+                channelId)
+                .setContentTitle(title)
+                .setContentText(body)
+                .setTicker("Droidcon Boston")
```
Comment (kenyee) [inline Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/NotificationUtils.java]: should this be in strings.xml?

File: Droidcon-Boston/app/build.gradle [hunk 4]
```diff
@@ -52,6 +52,8 @@ dependencies {
   implementation "com.android.support:cardview-v7:$support"
   implementation "com.android.support:design:$support"
 
+  // Firebase cloud messaging
+  implementation "com.google.firebase:firebase-messaging:$playServices"
```
Comment (kenyee) [inline Droidcon-Boston/app/build.gradle]: Doesn't this file need

apply plugin: 'com.google.gms.google-services'

at the end of it?  I'm trying to add Firebase DB and I think it needs this...

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java (hunk 1 above)
Comment (emezias) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java]: Proguard can take out Log.d. Let's skip Timber and use proguard for this point in a later pull request.

File: Droidcon-Boston/app/build.gradle (hunk 4 above)
Comment (kenyee) [reply Droidcon-Boston/app/build.gradle]: ugh....google_service.json is also in the wrong directory...it should be in the app directory :-P
I can move it my PR if you want or move it in yours if you need it.
Reading JSON in Java looks like it'll be a PITA :-(

Comment (emezias) [reply Droidcon-Boston/app/build.gradle]: Does this pull request have a DB? It's all GCM and notifications. Looks like @kenyee can push a branch that will fix the gradle with Proguard to drop Log.d and add in the additional play services for the FB Real-Time Database.

Comment (emezias) [reply Droidcon-Boston/app/build.gradle]: @kenyee - if you don't like pita then export to a csv and reuse the sqlite code from last year for the local data.

Comment (nisrulz) [reply Droidcon-Boston/app/build.gradle]: I missed the google-services plugin at the bottom. Was my new PR after the discussion to close the last one. 
I fixed it, tested the app for notification, updated the proguard-rules, moved the strings to strings.xml.
About adding realtime db, I was thinking others were working on it and would add it in their own PR. Its not added here

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java (hunk 1 above)
Comment (nisrulz) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java]: I added it in the last commit I pushed.

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/NotificationUtils.java (hunk 3 above)
Comment (nisrulz) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/NotificationUtils.java]: I added it in the last commit I pushed.

**Pull request merged**